agent-protocol.md

```diff
@@ -0,0 +1,224 @@
+# SMP agent protocol - duplex communication over SMP protocol
+
+## Table of contents
+
+- [Abstract](#abstract)
+- [SMP Agent](#smp-agent)
+- [SMP agent protocol components](#smp-agent-protocol-components)
+- [Duplex connection](#duplex-connection)
+- [Communication between SMP agents](#communication-between-smp-agents)
+  - [Message syntax](#messages-between-smp-agents)
+- [SMP agent commands](#smp-agent-commands)
+- [Connection invitation](#connection-invitation)
+
+## Abstract
+
+The purpose of SMP agent protocol is to define the syntax and the semantics of communications between the client and the agent that connects to [SMP](./simplex-messaging.md) servers.
+
+It provides:
+- convenient protocol to create and manage bi-directional (duplex) connection to the users of SMP agents consisting of two separate unidirectional (simplex) SMP queues, abstracting away multiple steps required to establish bi-directional connections.
+- management of E2E encryption between SMP agents, generating ephemeral RSA keys for each connection.
+- SMP command authentication on SMP servers, generating ephemeral RSA keys for each SMP queue.
+- TCP transport handshake and encryption with SMP servers.
+- validation of message integrity.
+
+SMP agent protocols provides no encryption or any security on the client side - it is assumed that the agent is executed in the trusted and secure environment.
+
+The future versions of this protocol could provide:
+- managing redundant SMP queues with more than 1 queue in each direction.
```

Comment:
add "message blockchain validation"? I don't remember if it was discussed as a part of this or SMP protocol..

Reply:
we need to add it - it is included in the current protocol version, but not implemented yet